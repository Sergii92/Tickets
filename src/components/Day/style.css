.day {
  width: 90%;
  display: flex;
  background-color: #cccccc;
  color: white;
  border-radius: 20px;
  max-width: 1300px;
  min-height: 150px;
  margin-bottom: 20px;
  padding: 30px;
  box-sizing: border-box;
  justify-content: space-around;
}
