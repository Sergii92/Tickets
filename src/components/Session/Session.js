import React from 'react'
import styled from 'styled-components';

import img from '../../assets/img/img.png'


const SessionWrapper = styled.div`
border: 0px solid rgba(0, 0, 0, 0);
border-radius: 5px;
width:33%;
max-width:300px;
border:1px solid black;
padding: 20px;
text-align:center;
background-color:black;
color:white


`
const SessionImageBlock = styled.div`
width:95%;
background-image:url(${img});
background-position:center;
background-repeat:no-repeat;

border-radius:10px;
height:300px;
margin:0 auto;
background-color:#9d9d9d;
margin-bottom:10px;


`


 const Session = () => {
  return (
  <SessionWrapper> 
    <SessionImageBlock/> 
   <h2>SessionName</h2>
  <h3>SessionTime</h3>

  </SessionWrapper>
  )
}


export default Session