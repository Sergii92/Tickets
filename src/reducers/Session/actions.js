import { TYPES } from "./types";

export const dataActions = {
  fetchData: () => ({
    type: TYPES.FETCH_DAYS,
  }),
  fetchDataSuccess: (data) => ({
    type: TYPES.FETCH_DATA_SUCCESS,
    payload: data,
  }),
  getSessionID: (id) => ({
    type: TYPES.GET_SESSION_ID,
    payload: id,
  }),
  fetchSession: () => ({
    type: TYPES.FETCH_SESSIONS,
  }),
  fetchSessionsSuccess: (data) => ({
    type: TYPES.FETCH_SESSIONS_SUCCESS,
    payload: data,
  }),

  fetchPlacesSuccess: (data) => ({
    type: TYPES.FETCH_PLACES,
    payload: data,
  }),
};
