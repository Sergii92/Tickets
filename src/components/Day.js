import React from "react";

import Session from "./Session";
import { DayBlock, Block, Title, SessionsBlock } from "../styled/components";

const Day = ({ item }) => {
  return (
    <Block>
      <DayBlock>
        <Title>{item.dayName}</Title>
      </DayBlock>
      <SessionsBlock>
        {item &&
          item.sessions.map((sesssion) => (
            <Session key={sesssion.id} sesssion={sesssion} />
          ))}
      </SessionsBlock>
    </Block>
  );
};

export default Day;
