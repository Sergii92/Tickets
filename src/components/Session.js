import React, { useState } from "react";
import SessionPage from "../pages/SessionPage";
import { SessionTitle, SessionBlock } from "../styled/components";
import { history } from "../constants/history";

const Session = ({ sesssion }) => {
  const [modalIsOpen, setModalIsOpen] = useState(false);

  const openModal = (e) => {
    history.push(`/session`);
    setModalIsOpen(!modalIsOpen);
    localStorage.setItem("modal", sesssion.name);
  };

  const closeModal = () => {
    setModalIsOpen(!modalIsOpen);

    history.push(`/home`);
    localStorage.clear();
  };

  return (
    <>
      <SessionBlock onClick={openModal}>
        <SessionTitle>{sesssion.time}</SessionTitle>
        <SessionTitle>{sesssion.name}</SessionTitle>
        <SessionTitle>
          Free places:{sesssion.totalPlaces - sesssion.booked}
        </SessionTitle>
        <SessionTitle>Booked places:{sesssion.booked}</SessionTitle>
      </SessionBlock>
      <SessionPage
        sesssion={sesssion}
        modalIsOpen={modalIsOpen}
        setModalIsOpen={setModalIsOpen}
        closeModal={closeModal}
      />
    </>
  );
};

export default Session;
