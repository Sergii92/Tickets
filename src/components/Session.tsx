import React, { useState } from "react";
import { useDispatch } from "react-redux";
import { Card } from "react-bootstrap";

import { CustomCard } from "../styled/components";
import MyModal from "./Modal";

import { dataActions } from "../redux/ducks/session";
import { history } from "../constants/history";
import { IsessionProps } from "../interfaces";
import SingleCard from "./SingleCard";

const Session = ({ sesssion }: IsessionProps) => {
  const dispatch = useDispatch();

  const [show, setShow] = useState(false);

  const handleClose = () => {
    setShow(false);
    history.push(`/home`);
  };
  const handleShow = (id: number) => {
    setShow(true);

    dispatch(dataActions.fetchPlaces());
    history.push(`/sesssion${id}`);
  };

  return (
    <>
      <SingleCard
        id={sesssion.id}
        sessionName={sesssion.sessionName}
        time={sesssion.time}
        onClick={() => handleShow(sesssion.id)}
      />
      <MyModal handleClose={handleClose} show={show} sesssion={sesssion} />
    </>
  );
};

export default Session;
