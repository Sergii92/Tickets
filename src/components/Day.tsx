import React, { useState } from "react";
import { useSelector, useDispatch } from "react-redux";
import { Accordion, Card, CardDeck } from "react-bootstrap";

import Session from "./Session";

import { dataActions, selectSessions } from "../redux/ducks/session";

interface ComponentProps {
  day?: any;
}

const Day = ({ day }: ComponentProps) => {
  const [isOpen, setIsOpen] = useState(false);

  const dispatch = useDispatch();

  const open = () => {
    if (isOpen) {
      setIsOpen(!isOpen);
    } else {
      setIsOpen(!isOpen);
      dispatch(dataActions.fetchSession());
    }
  };

  const sessionsData: Array<any> = useSelector(selectSessions);

  return (
    <Accordion>
      <Card onClick={() => open()}>
        <Accordion.Toggle as={Card.Header} eventKey={day.id}>
          {day.name}
        </Accordion.Toggle>
        <Accordion.Collapse eventKey={day.id}>
          <Card.Body>
            <CardDeck>
              {sessionsData &&
                sessionsData
                  .filter((item) => item.daysId === day.id)
                  .map((sesssion) => (
                    <Session key={sesssion.id} sesssion={sesssion} />
                  ))}
            </CardDeck>
          </Card.Body>
        </Accordion.Collapse>
      </Card>
    </Accordion>
  );
};

export default Day;
