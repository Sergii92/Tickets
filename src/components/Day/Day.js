import React from "react";

import styled from "styled-components";
import Session from "../Session/Session";

import "./style.css";

const DayBlock = styled.div`
  min-width: 120px;

  padding: 0 10px;
  background-color: darkslategrey;
  text-align: center;
`;

const Title = styled.p`
  color: white;
  font-size: 25px;
`;
const SessionsBlock = styled.div`
  width: 1000px;
  display: flex;
  heigth: auto;
  flex-wrap: wrap;
  justify-content: space-around;
`;

const Day = ({ item }) => {
  console.log(item);
  return (
    <div className="day">
      <DayBlock>
        <Title>{item.dayName}</Title>
      </DayBlock>
      <SessionsBlock>
        {item &&
          item.sessions.map((sesssion) => {
            return <Session key={sesssion.id} sesssion={sesssion} />;
          })}
      </SessionsBlock>
    </div>
  );
};

export default Day;
