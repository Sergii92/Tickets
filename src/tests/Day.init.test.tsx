import React from "react";
import { shallow } from "enzyme";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

import Day from "../components/Day";
import Session from "../components/Session";
import { mock } from "./testMock";
const mockStore = configureStore();

describe("Day Component", () => {
  const props = {
    day: { id: 1, name: "Test" },
    isOpen: true,
  };

  const store = mockStore(mock);

  describe.skip("Day component render", () => {
    it("RENDER", () => {
      const component = shallow(
        <Provider store={store}>
          <Day day={props.day} />
        </Provider>
      );
      expect(component).toMatchSnapshot();
    });

    it("Test no cards before open day block", () => {
      const component = shallow(
        <Provider store={store}>
          <Day day={props.day} />
        </Provider>
      );

      expect(component.find(Session)).toHaveLength(0);
    });
  });
});
