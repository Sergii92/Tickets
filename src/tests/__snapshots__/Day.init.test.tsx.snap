// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Day Component Day component render RENDER 1`] = `ShallowWrapper {}`;
